examples/api-tests/src/typescript.spec.js

```diff
@@ -0,0 +1,107 @@
+    const { HostedPluginSupport } = require('@theia/plugin-ext/lib/hosted/browser/hosted-plugin');
+    const { ContextKeyService } = require('@theia/core/lib/browser/context-key-service');
+    const { CommandRegistry } = require('@theia/core/lib/common/command');
+
+    /** @type {import('inversify').Container} */
+    const container = window['theia'].container;
+    const editorManager = container.get(EditorManager);
+    const workspaceService = container.get(WorkspaceService);
+    const menuFactory = container.get(BrowserMainMenuFactory);
+    const pluginService = container.get(HostedPluginSupport);
+    const contextKeyService = container.get(ContextKeyService);
+    const commands = container.get(CommandRegistry);
+
+    const rootUri = new Uri.default(workspaceService.tryGetRoots()[0].uri);
+    const serverUri = rootUri.resolve('src-gen/backend/server.js');
+    const inversifyUri = rootUri.resolve('../../node_modules/inversify/dts/inversify.d.ts').normalizePath();
+
+    /** @type {EditorWidget} */
+    let widget;
+    /** @type {MonacoEditor} */
+    let editor;
+
+    before(async function () {
+        this.timeout(5000);
+        await Promise.all([
+            pluginService.load(),
+            editorManager.closeAll({ save: false })
+        ]);
+        await Promise.all([
+            (async () => {
+                const plugin = pluginService.plugins.find(p => p.model.id === 'vscode.typescript-language-features');
+                await pluginService.activatePlugin(plugin.model.id);
+            })(),
+            (async () => {
+                widget = await editorManager.open(serverUri, { mode: 'activate' });
+                editor = MonacoEditor.get(widget);
+            })()
+        ]);
+        // wait till tsserver is running, see:
+        // https://github.com/microsoft/vscode/blob/93cbbc5cae50e9f5f5046343c751b6d010468200/extensions/typescript-language-features/src/extension.ts#L98-L103
+        await new Promise(resolve => {
+            if (contextKeyService.match('typescript.isManagedFile')) {
+                resolve();
+                return;
+            }
+            contextKeyService.onDidChange(() => {
+                if (contextKeyService.match('typescript.isManagedFile')) {
+                    resolve();
+                }
+            });
+        });
+    });
+
+    after(async () => {
+        widget = undefined;
+        editor = undefined;
+        await editorManager.closeAll({ save: false });
+    });
+
+    it('document formating should be visible and enabled', () => {
+        const menu = menuFactory.createContextMenu(EDITOR_CONTEXT_MENU);
+        const item = menu.items.find(i => i.command === 'editor.action.formatDocument');
+        assert.isDefined(item);
+        assert.isTrue(item.isVisible);
+        assert.isTrue(item.isEnabled);
+    });
+
+    it('reveal definition', async function () {
+        // const { Cont|ainer } = require('inversify');
```

Comment:
I tried it for the PR https://github.com/eclipse-theia/theia/pull/7149
This test fails by timeout for me.
So locally I had to add  `this.timeout(4000);` to have this test successful

Reply:
yes, i set it to 15s, that's too much if we are going to have many tests. We need some approach which start TS server only once, but still keeps tests isolated.